Bio/PDB/Mutate.py

```diff
@@ -0,0 +1,561 @@
+    "ARG": ["N", "CA", "C", "O", "CB", "CG", "CD", "NE", "CZ", "NH1", "NH2"],
+    "TRP": [
+        "N",
+        "CA",
+        "C",
+        "O",
+        "CB",
+        "CG",
+        "CD1",
+        "CD2",
+        "CE2",
+        "CE3",
+        "NE1",
+        "CZ2",
+        "CZ3",
+        "CH2",
+    ],
+    "ALA": ["N", "CA", "C", "O", "CB"],
+    "VAL": ["N", "CA", "C", "O", "CB", "CG1", "CG2"],
+    "GLU": ["N", "CA", "C", "O", "CB", "CG", "CD", "OE1", "OE2"],
+    "TYR": ["N", "CA", "C", "O", "CB", "CG", "CD1", "CD2", "CE1", "CE2", "CZ", "OH"],
+    "MET": ["N", "CA", "C", "O", "CB", "CG", "SD", "CE"],
+}
+
+
+def load_rotamers(rotamer_loc=DUNBRACK_FILTERED):
+    """
+    Load the Dunbrack rotamer library
+    """
+    _dunbrack = defaultdict(lambda: defaultdict(lambda: defaultdict(list)))
+    with gzip.open(rotamer_loc, "rt") as fn:
+        for line in fn:
+            if line.startswith("#"):
+                continue
+            amino_acid, angle1, angle2, prob, chi1, chi2, chi3, chi4 = line.split()
+            _dunbrack[amino_acid][int(angle1)][int(angle2)].append(
+                {
+                    "prob": float(prob),
+                    "CHI1": float(chi1),
+                    "CHI2": float(chi2),
+                    "CHI3": float(chi3),
+                    "CHI4": float(chi4),
+                }
+            )
+    return _dunbrack
+
+
+def rotation_matrix(axis, theta):
+    """
+    Calculates a rotational matrix around an axis with an angle theta
+    """
+    axis = np.asarray(axis)
+    axis = axis / np.linalg.norm(axis)
+
+    cos_theta = np.cos(theta)
+    sin_theta = np.sin(theta)
+
+    outer_product = np.outer(axis, axis)
+
+    cross_product_matrix = np.array(
+        [[0, -axis[2], axis[1]], [axis[2], 0, -axis[0]], [-axis[1], axis[0], 0]]
+    )
+
+    rotation_mat = (
+        cos_theta * np.eye(3)
+        + sin_theta * cross_product_matrix
+        + (1 - cos_theta) * outer_product
+    )
+
+    return rotation_mat
+
+
+def dihedral_from_vectors(v1, v2, v3, v4):
+    """Praxeolitic formula
+    1 sqrt, 1 cross product"""
+    b0 = -1.0 * (v2 - v1)
+    b1 = v3 - v2
+    b2 = v4 - v3
+
+    b1 /= np.linalg.norm(b1)
+
+    v = b0 - np.dot(b0, b1) * b1
+    w = b2 - np.dot(b2, b1) * b1
+
+    x = np.dot(v, w)
+    y = np.dot(np.cross(b1, v), w)
+    return np.arctan2(y, x)
+
+
+def distance(x, y):
+    """
+    Calculates the euclidian distance in 3D space
+    """
+    return np.sqrt((x[0] - y[0]) ** 2 + (x[1] - y[1]) ** 2 + (x[2] - y[2]) ** 2)
+
+
+def read_sample_residue(residue_name):
+    """
+    Reads a sample PDB residue
+    """
+    sample_residue = {}
+    with open(f"{DATA_DIR}/{residue_name.upper()}.pdb") as fn:
+        for line in fn:
+            sample_residue[line[12:16].strip()] = np.array(
+                [float(line[30:38]), float(line[38:46]), float(line[46:54])]
+            )
+    return sample_residue
+
+
+def is_backbone(atom):
```

Comment:
This method would belong in `Atom.py`, with a check that the residue name belongs to a standard aminoacid.

Reply:
You also want to make the search in a `set`, not a list.